src/handlers/gestures/gestureBuilder.ts

```diff
@@ -0,0 +1,126 @@
+import { SimpleGesture } from './simpleGestures';
+
+enum Relation {
+  Simultaneous,
+  Exclusive,
+  After,
+  RequireToFail,
+}
+
+type PendingGesture = {
+  relation: Relation;
+  gesture: SimpleGesture;
+};
+
+export class GestureBuilder {
+  private log: Array<PendingGesture> = [];
+
+  constructor(base: SimpleGesture) {
+    this.log.push({ relation: Relation.Exclusive, gesture: base });
+  }
+
+  simultaneousWith(gesture: SimpleGesture): GestureBuilder {
+    this.log.push({ relation: Relation.Simultaneous, gesture: gesture });
+
+    return this;
+  }
+
+  exclusiveWith(gesture: SimpleGesture): GestureBuilder {
+    this.log.push({ relation: Relation.Exclusive, gesture: gesture });
+
+    return this;
+  }
+
+  after(gesture: SimpleGesture): GestureBuilder {
+    this.log.push({ relation: Relation.After, gesture: gesture });
+
+    return this;
+  }
+
+  requireToFail(gesture: SimpleGesture): GestureBuilder {
+    this.log.push({ relation: Relation.RequireToFail, gesture: gesture });
+
+    return this;
+  }
+
+  build(): BuiltGesture {
+    let result = new BuiltGesture(this.prepare);
```

Comment:
```suggestion
    const result = new BuiltGesture(this.prepare);
```

Reply:
Fixed in fdabf483d5407f3ea1c8fb6f0c75309a533efe41